File: Java/math/Prime.java
// 素数クラス
class Prime {
    // O(√n): 素数判定を行うメソッド
    public static boolean isPrime(long n) {
        if (n < 2L) {
            return false;
        }
        long val = 2L;
        while (val * val <= n) {
            if (n % val == 0L) {
                return false;
            }
            val += 1L;
        }
        return true;
    }
}

File: math/prime/ruby/prime.rb
# Shunta の自作ライブラリ
# https://github.com/NAVYSHUNTA/atcoder-shunta-library/blob/main/math/prime/ruby/prime.rb

# O(√n): 素数判定
def prime?(n)
  return false if n < 2
  return true if n == 2
  return false if n.even?

  _skip_even = 2
  div = 3
  while div * div <= n
    if n % div == 0
      return false
    end
    div += _skip_even
  end

  true
end
